zapcore/core.go

```diff
@@ -42,6 +42,8 @@ type Core interface {
 	Write(Entry, []Field) error
 	// Sync flushes buffered logs (if any).
 	Sync() error
+	// Close cleans all WriterSync resources (if any).
+	Close() error
```

Comment:
we can't change the interface here, as it would be a backwards incompatible change -- existing core implementations without `Close()` will break if we make this change.

we can either:
 - return a separate exported type from `Buffer` that implements `WriteSyncer` but also adds `Close()` and document that the caller must call `Close()` when they're done
 - return `(WriteSyncer, io.Closer, error)` from the call to buffer

We could also return just `WriteSyncer` and document that there's a close method if the returned value is up-casted, but I'd prefer to have the type-system call out the list of methods explicitly.

Reply:
Good idea, I prefered this one like context:
```
buffer, cancel := Buffer(...)
defer cancel()
```

Reply:
updated @prashantv